include/boost/graph/adj_list_serialize.hpp

```diff
@@ -43,17 +43,18 @@ namespace serialization
         const unsigned int /* file_version */
     )
     {
-        typedef adjacency_list< OEL, VL, D, VP, EP, GP, EL > Graph;
-        typedef typename graph_traits< Graph >::vertex_descriptor Vertex;
+        using Graph = adjacency_list< OEL, VL, D, VP, EP, GP, EL >;
+        using Vertex = typename graph_traits< Graph >::vertex_descriptor;
+        using SerializedInteger = unsigned int;
```

Comment:
Isn't this going to cause a problem for users with, however unlikely, graphs where `vertices_size_type` is `std::size_t` and actually have billions or trillions or vertices?

Also, it would waste space for users that have billions of tiny graphs where they have customized `vertices_size_type` to be `unsigned char`, etc?